pandas/core/arrays/sparse/array.py

```diff
@@ -547,6 +547,11 @@ def from_spmatrix(cls, data: spmatrix) -> Self:
     def __array__(
         self, dtype: NpDtype | None = None, copy: bool | None = None
     ) -> np.ndarray:
+        if copy is False:
+            raise ValueError(
+                "Unable to avoid copy while creating an array as requested."
+            )
+
         fill_value = self.fill_value
 
         if self.sp_index.ngaps == 0:
```

Comment:
corner case, but also this branch is returning underlying values and so is not honoring a `copy=True`